draft-ietf-dnsop-domain-verification-techniques.md

```diff
@@ -215,33 +236,45 @@ Note that this is semantically the same as:
 
 The user SHOULD de-provision the resource record provisioned for DNS-based domain control validation once it is no longer required.
 
-## CNAME Record
+## CNAME Records
 
-CNAME records MAY be used instead of TXT records, but they should not be placed at the same domain name that is being validated. This is for the same reason already cited in {{pitfalls}}. CNAME records cannot co-exist with other data, and there may already be other record types that exist at the domain name. Instead, as with the TXT record recommendation, an application specific label should be added as a subdomain of the domain to be verified. This ensures that the CNAME does not collide with other record types. In practice, many application services that employ CNAMEs today use a random subdomain label, which also works to avoid collisions. But adding an application specific component makes it easier for the domain owner to keep track of why and for what service a validation record has been deployed.
+CNAME records MAY be used instead of TXT records, either for Delegated Domain Control Validation ({{delegated}}) or where specified by providers to support users who are unable to create TXT records.
+
+A provider supporting CNAME records MUST specify the use of a label (e.g., `_foo-challenge`) as a CNAME MUST NOT be placed at the same domain name that is being validated. This is for the same reason already cited in {{pitfalls}}. CNAME records cannot co-exist with other data, and there may already be other record types that exist at the domain name. Instead, as with the TXT record recommendation, a provider specific label should be added as a subdomain of the domain to be verified. This ensures that the CNAME does not collide with other record types.
```

Comment:
I think "MUST" specify the use of an application specific label is a good idea, and conforms to our general recommendations for TXT too. However, we should note that there are other providers that use CNAME and don't do this. For example Amazon Certificate Manger (ACM). to quote the example we cite in the appendix:

_<random-token1>.example.com.   IN   CNAME _<random-token2>.acm-validations.aws.

Here, their use of a random token is one way they avoid the collisions with other records at the same domain name.

Reply:
```suggestion
A provider supporting CNAME records MUST specify the use of a label (e.g., "\_foo-<token>" or even the less-recommended "\_<token>") as a CNAME MUST NOT be placed at the same domain name that is being validated. This is for the same reason already cited in {{pitfalls}}. CNAME records cannot co-exist with other data, and there may already be other record types that exist at the domain name. Instead, as with the TXT record recommendation, a provider specific label should be added as a subdomain of the domain to be verified. This ensures that the CNAME does not collide with other record types.
```

How about this clarification?